src/sentry/workflow_engine/processors/delayed_workflow.py

```diff
@@ -0,0 +1,398 @@
+            for condition in slow_conditions:
+                unique_queries = generate_unique_queries(
+                    condition, workflows_to_envs[dcg_to_workflow[dcg.id]]
+                )
+                query_values = [
+                    condition_group_results[unique_query][group_id]
+                    for unique_query in unique_queries
+                ]
+
+                try:
+                    handler = condition_handler_registry.get(condition.type)
+                except NoRegistrationExistsError:
+                    logger.exception(
+                        "No registration exists for condition",
+                        extra={"type": condition.type, "id": condition.id},
+                    )
+                    continue
+
+                if not handler.evaluate_value(query_values):
+                    continue
+
+                if action_match == "any":
+                    groups_to_fire[group_id].add(dcg)
+                    break
+                elif action_match == "all":
+                    conditions_matched += 1
+
+            if action_match == "all" and conditions_matched == len(slow_conditions):
+                groups_to_fire[group_id].add(dcg)
+    return groups_to_fire
+
+
+def split_dcgs(
+    data_condition_groups: list[DataConditionGroup], dcg_to_workflow: dict[int, int]
+) -> tuple[set[DataConditionGroup], set[DataConditionGroup]]:
+    WHEN_dcgs = set()
+    IF_dcgs = set()
+    for dcg in data_condition_groups:
+        workflow_id = dcg_to_workflow[dcg.id]
+        if WorkflowDataConditionGroup.objects.filter(
+            workflow_id=workflow_id, data_condition_group=dcg
+        ).exists():
+            IF_dcgs.add(dcg)
+        else:
+            WHEN_dcgs.add(dcg)
+    return WHEN_dcgs, IF_dcgs
+
+
+def fire_dcgs(
+    groups_to_fire: DefaultDict[int, set[DataConditionGroup]],
+    dcg_to_workflow: dict[int, int],
+    dcg_group_to_event_data: dict[tuple[int, int], dict[str, str]],
+    project: Project,
+) -> None:
+    dcg_sets = groups_to_fire.values()
+    dcgs = set()
+    for dcg_set in dcg_sets:
+        dcgs.update(dcg_set)
+
+    WHEN_dcgs, IF_dcgs = split_dcgs(dcgs, dcg_to_workflow)
+
+    group_to_groupevent = get_group_to_groupevent(
+        dcg_group_to_event_data, project.id, groups_to_fire.keys()
+    )
+    for group, group_event in group_to_groupevent.items():
+        job = WorkflowJob({"event": group_event})
+        detector = get_detector_by_event(job)
+
+        dcgs_to_fire = groups_to_fire[group.id]
+        group_WHEN_dcgs = WHEN_dcgs.intersection(dcgs_to_fire)
+        group_IF_dcgs = IF_dcgs.intersection(dcgs_to_fire)
+
+        filtered_actions: list[Action] = []
+        workflows = Workflow.objects.filter(when_condition_group__in=group_WHEN_dcgs)
+
+        filtered_actions.extend(list(evaluate_workflow_action_filters(workflows, job)))
+        filtered_actions.extend(list(get_filtered_actions(group_IF_dcgs, job)))
```

Comment:
TODO: figure out what to do if there are WHEN dcgs with slow conditions are connected to IF dcgs with slow conditions 😵‍💫